### PR TITLE
Create UserPage, UserGrid, UserCard components and setup UserStore and UserActions

### LINKED ISSUE #479: Create StudentOutreach component
`CardHeader` component should be given "Outreach Information" title.

### LINKED ISSUE #491: Create user actions and stores alt infrastructure

### LINKED ISSUE #492: Create basic UserGrid, UserCard, and ResponsibilitiesCard

### LINKED ISSUE #493: Create UserShowSerializer for users show controller action
Should have a:
`has_many :responsibilities, serializer: ResponsibilityBaseSerializer`

### PR DESCRIPTION
**Pull request by unzunz**
Fix #492 
Fix #493 
Fix #491 
Fix #479

### CODE REVIEW

File: app/assets/javascripts/components/user/user_page.es6.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  componentDidMount() {
+    ProfileStore.listen(this._listener);
+    UserStore.listen(this._listener);
+    ProfileActions.fetchProfile();
+    UserActions.fetchUser(this.props.id);
+  }
+
+  componentWillUnmount() {
+    UserStore.unlisten(this._listener);
```
Comment (warrenshen) [inline app/assets/javascripts/components/user/user_page.es6.jsx]: Don't forget to also call `unlisten` on the `ProfileStore`.

File: app/assets/javascripts/constants/api_constants.es6.jsx [hunk 2]
```diff
@@ -54,6 +54,9 @@
         login: '/api/users/login',
         logout: '/api/users/logout',
         profile: '/api/users/profile',
+        show: function(id) {
```
Comment (warrenshen) [inline app/assets/javascripts/constants/api_constants.es6.jsx]: Nice nice.

File: app/assets/javascripts/constants/route_constants.es6.jsx [hunk 3]
```diff
@@ -43,6 +43,9 @@
     get users() {
       return {
         index: '/users',
+        show: function(id) {
+          return `/users/${id}`;
+        }
```
Comment (warrenshen) [inline app/assets/javascripts/constants/route_constants.es6.jsx]: Missing a comma after this `}`.

File: app/controllers/api/users_controller.rb [hunk 4]
```diff
@@ -15,6 +15,12 @@ def profile
     end
   end
 
+  def show
+    user = User.find params[:id]
+    current_user.create_visit('User', params[:id].to_i)
```
Comment (warrenshen) [inline app/controllers/api/users_controller.rb]: Niceeeeeee.

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<%= react_component('UserPage', {id: @id }) %>
```
Comment (warrenshen) [inline app/views/users/show.html.erb]: Missing a space before `id:`.

File: config/routes.rb [hunk 6]
```diff
@@ -11,7 +11,7 @@
   resources :forms, only: [:show]
   resources :students, only: [:index, :show]
   resources :schools, only: [:index, :show]
-  resources :users, only: [:index]
+  resources :users, only: [:index, :show]
```
Comment (warrenshen) [inline config/routes.rb]: Woot.

File: config/routes.rb [hunk 7]
```diff
@@ -37,7 +37,7 @@
         resources :comments, only: [:create, :index]
       end
     end
-    resources :users, only: [:index, :update]
+    resources :users, only: [:index, :show, :update]
```
Comment (warrenshen) [inline config/routes.rb]: I hope it's clear why we have to define two sets of `:show`s for `:users`!

Comment (unzunz) [inline config/routes.rb]: yep! one for `/api/users/` and `/users/`

**Pull request merged**